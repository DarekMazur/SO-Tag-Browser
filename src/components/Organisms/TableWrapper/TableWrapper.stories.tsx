import type { Meta, StoryObj } from '@storybook/react';
import TableWrapper from './TableWrapper.tsx';

const meta: Meta<typeof TableWrapper> = {
  tags: ['autodocs'],
  component: TableWrapper,
  title: 'Organisms/TableWrapper',
};

export default meta;

type Story = StoryObj<typeof TableWrapper>;

export const Default: Story = {};
