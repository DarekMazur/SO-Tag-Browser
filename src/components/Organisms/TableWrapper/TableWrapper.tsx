import { useTranslation } from 'react-i18next';
import TableHeader from '../../Molecules/TableHeader/TableHeader.tsx';
import TableProvider from '../../../providers/TableProvider.tsx';
import TableContent from '../TableContent/TableContent.tsx';
import { ITableHeader } from '../../../types/types.ts';
import { useGetTagsQuery } from '../../../store';
import LoadingStatus from '../../Molecules/LoadingStatus/LoadingStatus.tsx';

const TableWrapper = () => {
  const { t } = useTranslation();
  const headers: ITableHeader[] = [
    {
      label: '',
      name: 'order',
      isSortable: false,
    },
    {
      label: t('table.header.tag'),
      name: 'name',
      isSortable: true,
    },
    {
      label: t('table.header.count'),
      name: 'count',
      isSortable: true,
    },
    {
      label: '',
      name: 'link',
      isSortable: false,
    },
  ];

  const { data: tags, isLoading, error } = useGetTagsQuery();

  return (
    <>
      {error ? (
        <LoadingStatus status="error" />
      ) : isLoading && !tags ? (
        <LoadingStatus status="loading" />
      ) : (
        <TableProvider>
          <TableHeader headers={headers} />
          <TableContent />
        </TableProvider>
      )}
    </>
  );
};

export default TableWrapper;
