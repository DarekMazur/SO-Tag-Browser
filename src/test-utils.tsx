// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import AppProviders from './providers/AppProviders.tsx';

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <AppProviders>{children}</AppProviders>;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customRender = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

export { customRender as render };
