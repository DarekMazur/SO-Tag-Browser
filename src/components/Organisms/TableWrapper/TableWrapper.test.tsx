import { render, screen, fireEvent } from '../../../test-utils';
import TableWrapper from './TableWrapper.tsx';

describe('Search Bar', () => {
  it('Renders the component', () => {
    const { container } = render(<TableWrapper />);
    expect(container).toMatchSnapshot();
  });
  it('Displays default items range in pagination', async () => {
    render(<TableWrapper />);
    await screen.findByText(/1â€“5/);
  });
  it('Displays total items in pagination', async () => {
    render(<TableWrapper />);
    await screen.findByText(/20/);
  });
  it('Displays next page after click in pagination', async () => {
    render(<TableWrapper />);
    const nextButton = screen.getByLabelText('next page');
    fireEvent.click(nextButton);
    await screen.findByText(/6â€“10/);
  });
  it('Displays table headers', async () => {
    render(<TableWrapper />);
    await screen.findByText(/Posts count/);
  });
});
