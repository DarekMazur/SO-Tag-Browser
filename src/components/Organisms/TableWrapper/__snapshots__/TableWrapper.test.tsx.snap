// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Bar > Renders the component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1njjnc6-MuiPaper-root"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-14v9whk-MuiCircularProgress-root"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-17psgs0-MuiTypography-root"
    >
      Loading...
    </p>
  </div>
</div>
`;
